README.md

```diff
@@ -1,155 +1,236 @@
+
+```dart
+class ToDo {
+  ToDo(this.toDoText);
+  String toDoText;
+  Bool isComplete = false;
+}
+
+class ToDoStore extends Store {
+  final _todos = new List<ToDo>();
+  List<ToDo> get todos => new List.unmodifiable(_todos);
+}
+```
+
+To handle an `Action`, a `Store` subscribes to the `Action` and defines a handler function, like this:
+
+```dart
+class ToDoStore extends Store {
+  ToDoStore() {
+    triggerOnAction(submitToDoAction, (InputValue value) {
+      _todos.add(new ToDo(value.text));
+    });
   }
+
+  final _todos = new List<ToDo>();
+  List<ToDo> get todos => new List.unmodifiable(_todos);
 }
 ```
 
-**BONUS:** `Stores` can be initialized with a stream transformer to modify the standard behavior of the `trigger` stream.
-This can be useful for throttling UI rendering in response to high frequency `Store` mutations.
+With this change, every time a widget dispatches `submitToDoAction`, the text of the new item will be added to the _todos list.
+
+You instantiate a store using a StoreToken, like this:
 
 ```dart
-import 'package:rate_limit/rate_limit.dart';
-import 'package:flutter_flux/flutter_flux.dart';
+final StoreToken todoStoreToken =
+    new StoreToken(new ToDoStore());
+```
+
+Hiding the `ToDoStore` inside of a `StoreToken` enforces the limited visibility of a `Store`. The `Store` will be explicitly passed to your `build()` function and can referenced by its token, as shown in the next section.
 
-class ThrottledStore extends Store {
-  ...
+We could add the trivial business logic to ignore empty ToDo items:
 
-  ThrottledStore(this._actions) : super.withTransformer(new Throttler(const Duration(milliseconds: 30))) {
-    ...
+```dart
+class ToDoStore extends Store {
+  ToDoStore() {
+    triggerOnAction(submitToDoAction, (InputValue value) {
+      if (value.text != null && value.text.trim().isNotEmpty) {
+        _todos.add(new ToDo(newvalue.text));
+      }
+    });
   }
+
+  final _todos = new List<ToDo>();
+  List<ToDo> get todos => new List.unmodifiable(_todos);
 }
 ```
 
-**BONUS:** `Stores` provide an optional terse syntax for action -> data mutation -> trigger operations.
+## **3. Widgets rebuild**
 
-```dart
-// verbose syntax
-actions.incrementCounter.listen(_handleAction);
+After the `Store` updates, the view (widgets) need to update to match the new model. A widget registers to listen to stores in its `initStores()` function. For example:
 
-_handleAction(payload) {
-    // perform data mutation
-    counter += payload;
-    trigger();
+```dart
+  @override
+  void initStores(ListenToStore listenToStore) {
+    listenToStore(todoStoreToken);
   }
+```
+
+Once you’ve made this call, then flutter_flux will automatically call `setState()` to force the tree to rebuild. Alternatively, you can pass a lambda function to `listenToStore()` if you want to define another behavior when a store announces a change. The important point is that, by default, you do not need to define an event listener or call `setState()`.
```

Comment:
Who or what is making this call? "Once you've made this call" is hard to understand.